### PR TITLE
Fix non-amd64 conformance image builds

### LINKED ISSUE #89554: Conformance image v1.18.0 fails on arm64
**What happened**:
`gcr.io/google-containers/conformance:v1.18.0` produces an error:
```
standard_init_linux.go:211: exec user process caused "exec format error"
```

**What you expected to happen**:
Conformance tests should be launched. Examining image contents indicates executables are of the correct architecture.

**How to reproduce it (as minimally and precisely as possible)**:
From aarch64 machine:
```
docker run --rm -it --entrypoint bash gcr.io/google-containers/conformance:v1.18.0 -c "echo ok"
```

**Anything else we need to know?**:
Works with v1.17.4:
```
# docker run --rm -it --entrypoint bash gcr.io/google-containers/conformance:v1.17.4 -c "echo ok"
ok
```

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0+k3s-5fe1f025", GitCommit:"5fe1f02592397c8a199b898113eb3e798905e918", GitTreeState:"clean", BuildDate:"2020-03-26T23:02:16Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/arm64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0+k3s-5fe1f025", GitCommit:"5fe1f02592397c8a199b898113eb3e798905e918", GitTreeState:"clean", BuildDate:"2020-03-26T23:02:16Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/arm64"}
```
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
```
Ubuntu 18.04.3 LTS
```
- Kernel (e.g. `uname -a`):
```
Linux ip-172-31-17-16 4.15.0-1054-aws #56-Ubuntu SMP Thu Nov 7 16:18:50 UTC 2019 aarch64 aarch64 aarch64 GNU/Linux
```
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by hasheddan**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

~Updates the conformance image to build for the correct architecture when
not amd64. This was previously implemented but caused issues in the
release pipeline due to an issue in buildkit with the baseimage not
being present in the gcr mirror registry. This change addresses that
concern by pre-pulling the baseimage.~

The approach above was abandoned in favor of specifying platform
for each build. This allows us to be more consistent with other image
builds in k/k. Note that this did require switching to `debian:stable-slim`
because `debian:stretch-slim` does not have support for s390x in its
multi-arch build.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

**Which issue(s) this PR fixes**:

Fixes #89554 

**Special notes for your reviewer**:

Based on work done initially in https://github.com/kubernetes/kubernetes/pull/90400 by @zhlhahaha then subsequently reverted in https://github.com/kubernetes/kubernetes/pull/92042.

See https://github.com/moby/buildkit/issues/1271 for more info on the issue and https://github.com/kubernetes-sigs/cluster-addons/pull/84 for an example of working around it.

/assign @BenTheElder @dims 
/cc @kubernetes/release-engineering 
/sig release
/milestone v1.21
/priority critical-urgent

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cluster/images/conformance/Makefile [hunk 1]
```diff
@@ -56,15 +59,19 @@ endif
...
+	docker pull $(BASEIMAGE)
 
-push: build
-	docker push ${REGISTRY}/conformance-${ARCH}:${VERSION}
+	DOCKER_CLI_EXPERIMENTAL=enabled docker buildx build --pull --platform linux/$(ARCH) $(OUTPUT) -t ${REGISTRY}/conformance-${ARCH}:${VERSION} ${TEMP_DIR}
 ifeq ($(ARCH),amd64)
 	docker rmi ${REGISTRY}/conformance:${VERSION} 2>/dev/null || true
-	docker tag ${REGISTRY}/conformance-${ARCH}:${VERSION} ${REGISTRY}/conformance:${VERSION}
-	docker push ${REGISTRY}/conformance:${VERSION}
+	DOCKER_CLI_EXPERIMENTAL=enabled docker buildx build --pull --platform linux/$(ARCH) $(OUTPUT) -t ${REGISTRY}/conformance:${VERSION} ${TEMP_DIR}
```
Comment (BenTheElder) [inline cluster/images/conformance/Makefile:70]: we need to ensure buildx is initialized before trying to build, a clean docker install on linux doesn't have a builder

Comment (hasheddan) [reply cluster/images/conformance/Makefile:70]: @BenTheElder won't it use docker driver by default?

Comment (hasheddan) [reply cluster/images/conformance/Makefile]: Resolving this as we are moving away from using buildkit here

**Comment by hasheddan** [comment]
@BenTheElder @dims @saschagrunert can y'all take another review pass here? :)

**Review by saschagrunert — APPROVED** [review]
/lgtm

I'll do a manual release stage with those changes to see if the conformance image gets build successfully:
https://console.cloud.google.com/cloud-build/builds/83f7e2b6-863b-4eec-9a73-c4f5a1b260b2?project=648026197307

**Comment by dims** [comment]
/approve
/hold 

( please feel free to remove hold when you feel this is right @hasheddan )

**Comment by hasheddan** [comment]
@saschagrunert looks like images were built successfully on that staging run :+1:

**Comment by saschagrunert** [comment]
Yes, this looks good.
/hold cancel

File: cluster/images/conformance/Dockerfile [hunk 2]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM BASEIMAGE
+FROM --platform=linux/ARCH BASEIMAGE
```
Comment (BenTheElder) [inline cluster/images/conformance/Dockerfile:15]: since when does docker support this?

Comment (BenTheElder) [reply cluster/images/conformance/Dockerfile:15]: ah https://github.com/kubernetes/kubernetes/pull/98569 anyhow 🙃

**Pull request merged**